### PR TITLE
Adds support for PyTorch

### PR DESCRIPTION
**Pull request by valentindey**
adds external configurables for PyTorch:
* losses
* optimizers
* learning rate schedulers
* activation functions
* random distributions
* constants

this is an updated version of #21 - credit goes to @georgepar

### CODE REVIEW

File: tests/torch/external_configurables_test.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            torch.optim.SGD,
+        ]
+        for optimizer in optimizers:
+            config.clear_config()
+            config_lines = ["""fake_train_model.optimizer = @{optimizer}
+                            {optimizer}.lr = 0.001"""
+                            .format(optimizer=optimizer.__name__)]
+            config.parse_config(config_lines)
+
+        # pylint: disable=no-value-for-parameter
```
Comment (sguada) [inline tests/torch/external_configurables_test.py]: This should be inside the loop, since I suppose you want to test that all the configs are valid optimizers.

Comment (valentindey) [reply tests/torch/external_configurables_test.py]: True, I updated it.

Made me also realize that I used a different indentation level than you guys. Updated that as well.

**Comment by valentindey** [comment]
ping @sguada

**Comment by dhr** [comment]
Thanks for the PR and sorry for the long delay in getting this in! I just merged this (had to make a few minor changes to make things compatible with internal tools and tests), so going to close the PR now.

**Comment by valentindey** [comment]
Great, thanks for merging! Could someone also please take care of updating the package on pypi?

**Comment by georgepar** [comment]
Thanks for merging guys. Special thanks to @valentindey for taking over :)
